### PR TITLE
In-app messages inbox

### PR DESCRIPTION
**Pull request by MattLichtenstein**
## Description

Implements in-app messaging inbox view

## Reference

[VPN-2488: https://mozilla-hub.atlassian.net/browse/VPN-2488](https://mozilla-hub.atlassian.net/browse/VPN-2488)

## Checklist
    
- [ ] My code follows the style guidelines for this project
- [ ] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [ ] I have performed a self review of my own code
- [ ] I have commented my code PARTICULARLY in hard to understand areas
- [ ] I have added thorough tests where needed

### CODE REVIEW

File: src/addons/addonmessage.h [hunk 1]
```diff
@@ -55,6 +59,8 @@ class AddonMessage final : public Addon {
   Composer* m_composer = nullptr;
 
   qint64 m_date = 0;
+  QString m_formattedDate = "";
+  QString m_subtitle = "";
```
Comment (bakulf) [inline src/addons/addonmessage.h:63]: Move these 2 next to m_title and remove `= ""`

File: src/addons/addonmessage.cpp [hunk 2]
```diff
@@ -88,17 +91,30 @@ void AddonMessage::maskAsRead() {
...
 
   QStringList readAddonMessages = settingsHolder->readAddonMessages();
   readAddonMessages.append(id());
   settingsHolder->setReadAddonMessages(readAddonMessages);
 
-  m_isRead = true;
   emit isReadChanged();
 }
 
+bool AddonMessage::containsSearchString(const QString& query) const {
```
Comment (bakulf) [inline src/addons/addonmessage.cpp:106]: `if query.isEmpty return true`

File: src/ui/screens/messaging/ViewMessagesInbox.qml [hunk 3]
```diff
@@ -0,0 +1,350 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.15
+import QtQuick.Controls 2.14
```
Comment (bakulf) [inline src/ui/screens/messaging/ViewMessagesInbox.qml:6]: This is a dup of src/ui/messages

**Review by bakulf — CHANGES_REQUESTED** [review]

File: translations/strings.yaml [hunk 4]
```diff
@@ -689,7 +689,15 @@ tutorialPopup:
   tutorialLeavePrimaryButtonLabel: Resume tutorial
 
 inAppMessaging:
+  menuTitle: Inbox
+  editButton: Edit
+  searchBarPlaceholderText: Search messages
   dateTimeYesterday: Yesterday
+  emptyStateTitle: No messages
+  emptyStateDescription: You're up to date!
```
Comment (flodolo) [inline translations/strings.yaml:697]: https://github.com/mozilla-mobile/mozilla-vpn-client/runs/8121313882?check_suite_focus=true

File: src/addons/addonmessage.cpp [hunk 5]
```diff
@@ -43,6 +43,8 @@ Addon* AddonMessage::create(QObject* parent, const QString& manifestFileName,
   message->m_title.initialize(QString("message.%1.title").arg(messageId),
                               messageObj["title"].toString());
 
+  message->m_subtitle = messageObj["subtitle"].toString();
```
Comment (bakulf) [inline src/addons/addonmessage.cpp:46]: can you use a AddonProperty? The code will look like the m_title.

File: src/addons/addonmessage.cpp [hunk 6]
```diff
@@ -88,17 +90,36 @@ void AddonMessage::maskAsRead() {
...
   QStringList readAddonMessages = settingsHolder->readAddonMessages();
   readAddonMessages.append(id());
   settingsHolder->setReadAddonMessages(readAddonMessages);
 
-  m_isRead = true;
   emit isReadChanged();
 }
 
+bool AddonMessage::containsSearchString(const QString& query) const {
+    if(query.isEmpty())
```
Comment (bakulf) [inline src/addons/addonmessage.cpp:106]: {}

File: src/addons/addonmessage.cpp [hunk 7]
```diff
@@ -88,17 +90,36 @@ void AddonMessage::maskAsRead() {
...
+        return true;
+
+    if(m_title.get().contains(query, Qt::CaseInsensitive))
```
Comment (bakulf) [inline src/addons/addonmessage.cpp:109]: {}

File: src/addons/addonmessage.cpp [hunk 8]
```diff
@@ -88,17 +90,36 @@ void AddonMessage::maskAsRead() {
...
+        return true;
+
+    if(m_subtitle.contains(query, Qt::CaseInsensitive))
+        return true;
+
+    for(ComposerBlock* block : m_composer->get()) {
+         if(block->contains(query))
```
Comment (bakulf) [inline src/addons/addonmessage.cpp:116]: {}

File: src/addons/addonmessage.cpp [hunk 9]
```diff
@@ -88,17 +90,36 @@ void AddonMessage::maskAsRead() {
...
 }
 
+bool AddonMessage::containsSearchString(const QString& query) const {
+    if(query.isEmpty())
+        return true;
+
+    if(m_title.get().contains(query, Qt::CaseInsensitive))
+        return true;
+
+    if(m_subtitle.contains(query, Qt::CaseInsensitive))
```
Comment (bakulf) [inline src/addons/addonmessage.cpp:112]: {}

File: src/addons/addonmessage.h [hunk 10]
```diff
@@ -19,7 +19,9 @@ class AddonMessage final : public Addon {
 
   Q_PROPERTY(Composer* composer MEMBER m_composer CONSTANT)
   Q_PROPERTY(bool isRead MEMBER m_isRead NOTIFY isReadChanged)
-  Q_PROPERTY(QString date READ date NOTIFY retranslationCompleted)
+  Q_PROPERTY(qint64 date MEMBER m_date)
+  Q_PROPERTY(QString formattedDate READ formattedDate NOTIFY retranslationCompleted)
+  Q_PROPERTY(QString subtitle MEMBER m_subtitle NOTIFY retranslationCompleted)
```
Comment (bakulf) [inline src/addons/addonmessage.h:24]: this is an ADDON_PROPERTY

File: src/composer/composerblockunorderedlist.cpp [hunk 11]
```diff
@@ -72,3 +72,12 @@ ComposerBlockUnorderedList::ComposerBlockUnorderedList(Composer* composer,
 ComposerBlockUnorderedList::~ComposerBlockUnorderedList() {
   MVPN_COUNT_DTOR(ComposerBlockUnorderedList);
 }
+
+bool ComposerBlockUnorderedList::contains(const QString& string) const {
+    for(const QString& item : m_subBlocks.get()) {
+        if(item.contains(string, Qt::CaseInsensitive))
```
Comment (bakulf) [inline src/composer/composerblockunorderedlist.cpp:78]: {}

**Review by bakulf — CHANGES_REQUESTED** [review]

File: src/addons/addonmessage.h [hunk 12]
```diff
@@ -52,6 +54,8 @@ class AddonMessage final : public Addon {
 
  private:
   AddonProperty m_title;
+  AddonProperty m_subtitle;
+  QString m_formattedDate;
```
Comment (bakulf) [inline src/addons/addonmessage.h:58]: remove this. You don't use it.

**Review by bakulf — APPROVED** [review]
1. remove m_formattedDate
2. you need to run clang-format

**Review by flodolo — APPROVED** [review]

**Pull request merged**